contracts/src/periphery/EigenDACertVerifierV1.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifierV1} from "src/periphery/interfaces/IEigenDACertVerifierV1.sol";
+import {IEigenDAThresholdRegistry} from "src/interfaces/IEigenDAThresholdRegistry.sol";
+import {IEigenDABatchMetadataStorage} from "src/interfaces/IEigenDABatchMetadataStorage.sol";
+import {EigenDACertVerificationV1Lib as CertLib} from "src/periphery/libraries/EigenDACertVerificationV1Lib.sol";
+import "src/interfaces/IEigenDAStructs.sol";
+
+/**
+ * @title CertVerifierV1 - EigenDA V1 certificate verification
+ * @author Layr Labs, Inc.
+ * @notice Contains all V1-specific verification functionality
+ */
+contract EigenDACertVerifierV1 is IEigenDACertVerifierV1 {
+    /// @notice Thrown when there is a length mismatch
+    error LengthMismatch();
+
+    /// @notice The EigenDAThresholdRegistry contract address
+    IEigenDAThresholdRegistry public immutable eigenDAThresholdRegistryV1;
+
+    /// @notice The EigenDABatchMetadataStorage contract address
+    /// @dev On L1 this contract is the EigenDA Service Manager contract
+    IEigenDABatchMetadataStorage public immutable eigenDABatchMetadataStorageV1;
```

Comment:
@samlaf #1472 

what is it on not L1? We don't know yet I'm guessing?

Reply:
I should probably remove that part, as there is nothing we don't know yet about how V1 will work, as it is going to be deprecated AFAICT

Reply:
https://github.com/Layr-Labs/eigenda/pull/1477/commits/96fda95576fc07c3045029a01d4df814b6eff3f8